core/devmode-spi/src/main/java/io/quarkus/dev/ErrorPageGenerators.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.dev;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * The generators can be used to generate a custom HTML page for a specific deployment exception that occurs during the
+ * development mode.
+ * <p>
+ * In order to avoid classloading issues the generators should not access the root cause directly but use reflection instead
+ * (the exception class could be loaded by a different class loader).
+ */
+public class ErrorPageGenerators {
```

Comment:
This is obviously useful, but I wonder if we should not put something more declarative in place, like `ExceptionMapper<T>` which JAX-RS has?

Reply:
I think that this could cause class loading issues...